### PR TITLE
[WIP] Add client for external-snapshotter api

### PR DESCRIPTION
**Pull request by saheienko**
Add a ops-like client for the csi snapshotter api (https://github.com/kubernetes-csi/external-snapshotter).

Signed-off-by: Serhii Aheienko <serhii.aheienko@gmail.com>

### CODE REVIEW

File: k8s/externalsnapshotter/externalsnapshotter.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+// SetConfig sets the config and resets the client
+func (c *Client) SetConfig(cfg *rest.Config) {
+	c.config = cfg
+	c.snap = nil
+}
+
+// initClient the k8s client if uninitialized
+func (c *Client) initClient() error {
+	if c.snap != nil {
```
Comment (piyush-nimbalkar) [inline k8s/externalsnapshotter/externalsnapshotter.go:87]: Do we even need `snap` variable? We should just be checking `if c.client != nil` and remove `snap` from Client.

File: k8s/externalsnapshotter/externalsnapshotter.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+)
+
+var (
+	instance Ops
+	once     sync.Once
+)
+
+// Ops is an interface to perform operations on external storage resources.
+type Ops interface {
+	SnapshotOps
```
Comment (piyush-nimbalkar) [inline k8s/externalsnapshotter/externalsnapshotter.go:20]: We need to add SnapshotContentOps and SnapshotClassOps here

File: k8s/externalsnapshotter/volumesnapshotclasses.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+// SnapshotClassOps is an interface to perform k8s VolumeSnapshotClass operations
+type SnapshotClassOps interface {
+	// CreateSnapshotClass creates the given snapshot
+	CreateSnapshotClass(snap *v1beta1.VolumeSnapshotClass) (*v1beta1.VolumeSnapshotClass, error)
+	// GetSnapshotClass returns the snapshot for given name
+	GetSnapshotClass(name string) (*v1beta1.VolumeSnapshotClass, error)
+	// ListSnapshotClasss lists all snapshots
+	ListSnapshotClasss() (*v1beta1.VolumeSnapshotClassList, error)
```
Comment (piyush-nimbalkar) [inline k8s/externalsnapshotter/volumesnapshotclasses.go:15]: nit: ListSnapshotClasses

File: k8s/externalsnapshotter/volumesnapshotclasses.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package externalsnapshotter
+
+import (
+	"github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SnapshotClassOps is an interface to perform k8s VolumeSnapshotClass operations
+type SnapshotClassOps interface {
+	// CreateSnapshotClass creates the given snapshot
```
Comment (piyush-nimbalkar) [inline k8s/externalsnapshotter/volumesnapshotclasses.go:10]: nit: creates the given snapshot class
(same for all the comments below)

File: k8s/externalsnapshotter/volumesnapshotcontents.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
+package externalsnapshotter
+
+import (
+	"github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SnapshotContentOps is an interface to perform k8s VolumeSnapshotContent operations
+type SnapshotContentOps interface {
+	// CreateSnapshotContent creates the given snapshot
```
Comment (piyush-nimbalkar) [inline k8s/externalsnapshotter/volumesnapshotcontents.go:10]: nit: .. given snapshot content
(same for all comments below)

File: k8s/externalsnapshotter/externalsnapshotter.go (hunk 1 above)
Comment (saheienko) [reply k8s/externalsnapshotter/externalsnapshotter.go]: Yes, there is no need for snap. Updated.

File: k8s/externalsnapshotter/externalsnapshotter.go (hunk 2 above)
Comment (saheienko) [reply k8s/externalsnapshotter/externalsnapshotter.go:20]: Updated.

File: k8s/externalsnapshotter/volumesnapshotclasses.go (hunk 3 above)
Comment (saheienko) [reply k8s/externalsnapshotter/volumesnapshotclasses.go]: Updated.

File: k8s/externalsnapshotter/volumesnapshotclasses.go (hunk 4 above)
Comment (saheienko) [reply k8s/externalsnapshotter/volumesnapshotclasses.go]: Updated.

File: k8s/externalsnapshotter/volumesnapshotcontents.go (hunk 5 above)
Comment (saheienko) [reply k8s/externalsnapshotter/volumesnapshotcontents.go]: Updated.

**Pull request merged**